qt: drop unused watch-only functionality

Sjors opened the pull request:
The watch-only functionality in the GUI was only used for legacy wallets. Watch-only descriptor wallets do not use this.

The only visible changes of this PR should be:
- dropped "Spendable:" label from the overview tab
- column width cache is reset

This PR also removes some unused variables from the interface.

hebasto commented on src/qt/bitcoinstrings.cpp:
```diff
@@ -317,12 +317,9 @@ QT_TRANSLATE_NOOP("bitcoin-core", "Error reading configuration file: %s"),
 QT_TRANSLATE_NOOP("bitcoin-core", "Error reading from database, shutting down."),
 QT_TRANSLATE_NOOP("bitcoin-core", "Error reading next record from wallet database"),
 QT_TRANSLATE_NOOP("bitcoin-core", "Error: Cannot extract destination from the generated scriptpubkey"),
-QT_TRANSLATE_NOOP("bitcoin-core", "Error: Could not add watchonly tx %s to watchonly wallet"),
```
https://github.com/bitcoin/bitcoin/blob/b070ce16966fe18eb68f886ab9ca02d1d05a2f2a/src/qt/bitcoinstrings.cpp#L5

Sjors replied:
Dropped the file.

fanquake commented:
Can be rebased now.

Sjors commented:
Somehow the Date column disappeared.

Sjors committed:
qt: drop unused watch-only functionality

The watch-only functionality in the GUI was only used for legacy wallets.
Watch-only descriptor wallets do not use this.

The only visible changes of this commit are:
- dropped "Spendable:" label from the overview tab
- column width cache is reset

hebasto reviewed (commented):
Are these lines still necessary:https://github.com/bitcoin/bitcoin/blob/59e09e0fb7b4cf8f8db97c2f81a8f6f69fe6cacf/src/interfaces/wallet.h#L374-L377 ?

Sjors committed:
wallet: drop watch-only things from interface

hebasto commented on src/qt/transactionview.h:
```diff
@@ -54,9 +54,9 @@ class TransactionView : public QWidget
-        WATCHONLY_COLUMN_WIDTH = 23,
         DATE_COLUMN_WIDTH = 120,
         TYPE_COLUMN_WIDTH = 113,
+        LABEL_MINIMUM_COLUMN_WIDTH = 120,
```
Where is this value used?

Sjors replied:
It's not, dropping it again.

Sjors commented:
> Are these lines still necessary:

I got rid of them.

For some reason QT is hiding column 1, which happens to be the date, but it also happens if I e.g. swap the date and type column.

hebasto commented:
> > Are these lines still necessary:
> 
> I got rid of them.
> 
> For some reason QT is hiding column 1, which happens to be the date, but it also happens if I e.g. swap the date and type column.
> 
> Aha, QT caches column layout between sessions, so it remembers how this column behaved on an earlier run.

^ Indeed.

Sjors commented:
I renamed `TransactionViewHeaderState` so that it's reset. This prevents hiding the date column.

hebasto commented:
> I renamed `TransactionViewHeaderState` so that it's reset. This prevents hiding the date column.

A more general approach would involve versioning, though.

Sjors commented:
Added a release note.

> A more general approach would involve versioning

I'm open to suggestions. Though this screen is unlikely to change again between now and when we switch to QML.

luke-jr commented:
Shouldn't this be on the GUI repo?

achow101 commented:
ACK 0fc9bf05c36169ff746df10c5c8310f5adf2e202

There does not appear to be any visual differences, nor any remaining references to watch-only/watchonly in the GUI.

hebasto reviewed (commented):
In `src/qt/forms/overviewpage.ui`, after removing all items from `row="0"` of the `gridLayout` element, all subsequent rows should be renumbered as follows: 1 -> 0, 2 -> 1, 3 -> 2, 4 -> 3, 5 -> 4.

Sjors commented:
Rebased just in case. Renumbered columns in `overviewpage.ui`.

hebasto reviewed (approved):
ACK e8661aac752eb08fee318eb8f56e599578d78f9f, I have reviewed the code and it looks OK. The `src/qt/forms/overviewpage.ui` form was reviewed in Qt Designer.

Tested on Ubuntu 25.04.

davidgumberg commented:
Review ACK https://github.com/bitcoin/bitcoin/commit/e8661aac752eb08fee318eb8f56e599578d78f9f. 

Quickly tested with GUI built on master and GUI built on this branch. Confirmed the watch-only behavior removed here has nothing to do with no-private-keys descriptor wallets, this is dead code AFAICT.
